spec/controllers/api/v1/links_controller_spec.rb

```diff
@@ -42,4 +42,58 @@
       end
     end
   end
+
+  describe "#show" do
+    context  "with invalid api_key" do
+      it "returns a 401" do
+        get :show, api_key: 'blank', id: link.short_url
+        expect(response.code).to eq '401'
+      end
+    end
+
+    context "with a valid api_key" do
+      let(:json) { JSON.parse(response.body) }
+
+      before do
+        get :show, api_key: user.api_key, id: link.short_url
+      end
+
+      it "returns a JSON response" do
```

Comment:
The problem is, the link doesn't belong to the user in this case! Right now, @link.user returns nil because the link was created by someone without a user account.  To fix this, go up to lines 5,6:

```
  let(:user) { create(:user) }
  let(:link) { create(:link, user_id: user.id) }
```
This assigns that link to the user, letting you call @link.user.name correctly!